Eclipse MAT moving to Java 11 and 2021-03/06

Eclipse MAT is moving to Java 11 and newer Eclipse Framework releases.

There are two distinct parts to this:
- Java 11, which should be easy
- Eclipse framework, which is a bit more tricky

A basic swap to Java 11 and Eclipse framework at 4.18, JIFA fails with NullPointerException related to `Platform.getContentTypeManager()`. Haven't looked any further other than it seems related to `Platform.getContentTypeManager()` being null.

```
Caused by: java.lang.NullPointerException
	at org.eclipse.mat.parser.internal.SnapshotFactoryImpl.openSnapshot(SnapshotFactoryImpl.java:94)
	at org.eclipse.mat.snapshot.SnapshotFactory.openSnapshot(SnapshotFactory.java:147)
	at org.eclipse.jifa.hda.impl.HeapDumpAnalyzerImpl.lambda$open$1(HeapDumpAnalyzerImpl.java:174)
	at org.eclipse.jifa.hda.impl.HeapDumpAnalyzerImpl.$(HeapDumpAnalyzerImpl.java:158)
```

--- a/backend/build.gradle
+++ b/backend/build.gradle
@@ -26,8 +26,8 @@ def HEAP_DUMP_ANALYZER_PROVIDER = 'heap-dump-analyzer:provider'
             apply plugin: 'java'
             compileJava {
                 options.encoding = "UTF-8"
-                targetCompatibility = JavaVersion.VERSION_1_8
-                sourceCompatibility = JavaVersion.VERSION_1_8
+                targetCompatibility = JavaVersion.VERSION_11
+                sourceCompatibility = JavaVersion.VERSION_11
                 options.compilerArgs << "-Xlint:deprecation"
             }
             test {

--- a/backend/heap-dump-analyzer/eclipse-mat-deps/build.gradle
+++ b/backend/heap-dump-analyzer/eclipse-mat-deps/build.gradle
@@ -17,14 +17,16 @@ plugins {
 
 p2AsMaven {
     group 'eclipse-deps', {
-        repoEclipse '4.8'
+        repoEclipse '4.18'
         iu 'com.ibm.icu'
+        iu 'org.apache.felix.scr'
         iu 'org.eclipse.core.commands'
         iu 'org.eclipse.core.contenttype'
         iu 'org.eclipse.core.jobs'
         iu 'org.eclipse.core.runtime'
         iu 'org.eclipse.equinox.app'
         iu 'org.eclipse.equinox.common'
+        iu 'org.eclipse.equinox.event'
         iu 'org.eclipse.equinox.preferences'
         iu 'org.eclipse.equinox.registry'
         iu 'org.eclipse.osgi'
@@ -35,7 +37,7 @@ p2AsMaven {
     }
 
     group 'eclipse-mat', {
-        repo 'http://download.eclipse.org/mat/1.11.0/update-site/'
+        repo 'http://download.eclipse.org/mat/1.12.0/update-site/'
         iu 'org.eclipse.mat.api'
         iu 'org.eclipse.mat.hprof'
         iu 'org.eclipse.mat.parser'

--- a/backend/heap-dump-analyzer/impl/build.gradle
+++ b/backend/heap-dump-analyzer/impl/build.gradle
@@ -18,7 +18,7 @@ jar {
     manifest {
         attributes(
                 "Bundle-SymbolicName": "org.eclipse.jifa.hda.implementation; singleton:=true",
-                "Require-Bundle": "org.eclipse.mat.api;bundle-version=\"1.10.0\",org.eclipse.mat.parser;bundle-version=\"1.10.0\",org.eclipse.osgi",
+                "Require-Bundle": "org.eclipse.mat.api;bundle-version=\"1.12.0\",org.eclipse.mat.parser;bundle-version=\"1.12.0\",org.eclipse.osgi",
                 "Bundle-Activator": "org.eclipse.jifa.hda.impl.Activator",
                 "Import-Package":
                         "org.eclipse.mat.snapshot,org.eclipse.mat.util,org.eclipse.mat.query,org.eclipse.mat.parser.model",
@@ -37,6 +37,8 @@ dependencies {
     implementation files(
             "${prefix}org.eclipse.osgi.jar",
             "${prefix}org.eclipse.equinox.registry.jar",
+            "${prefix}org.apache.felix.scr.jar",
+            "${prefix}org.eclipse.equinox.event.jar",
             "${prefix}org.eclipse.mat.report.jar",
             "${prefix}org.eclipse.mat.api.jar",
             "${prefix}org.eclipse.mat.parser.jar",

--- a/backend/worker/src/main/java/org/eclipse/jifa/worker/support/Analyzer.java
+++ b/backend/worker/src/main/java/org/eclipse/jifa/worker/support/Analyzer.java
@@ -129,6 +129,8 @@ public class Analyzer {
                         break;
                 }
             } catch (Exception e) {
+                LOGGER.error("task failed due to {}", ErrorUtil.toString(e));
+                LOGGER.error(progressListener.log());
                 File log = new File(FileSupport.errorLogPath(fileType, fileName));
                 FileUtil.write(log, progressListener.log(), false);
                 FileUtil.write(log, ErrorUtil.toString(e), true);

